homeassistant/components/mobile_app/__init__.py

```diff
@@ -82,7 +82,8 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
             )
 
     hass.async_create_task(
-        discovery.async_load_platform(hass, Platform.NOTIFY, DOMAIN, {}, config)
+        discovery.async_load_platform(hass, Platform.NOTIFY, DOMAIN, {}, config),
+        eager_start=True,
```

Comment:
According to the docstring of `async_load_platform` we need to use a task to avoid a deadlock. Is that not true anymore?

Reply:
Its unlikely that it will dead lock, and I don't think it can anymore, but I can't predict all paths that could happen with platform integrations so I can't say for sure either way.

Its much more likely finish synchronously now and never have to suspend, but it could take a long time because it could cause the base platform to be loaded which will do config validation of any platform integrations (ones listed as `platform` under it in `configuration.yaml`)

Its still better to do these in tasks so anything that depends the integration creating the tasks can proceed.

Reply:
Why can't we know all the places where a deadlock can happen? `discovery.async_listen_platform` isn't used in many places.

Here's the one that I've found:
https://github.com/home-assistant/core/blob/69a6e9f5d7c842371e5300084cc970c93c106557/homeassistant/setup.py#L528

Reply:
I don't think we can get there because it should already be setup by https://github.com/home-assistant/core/blob/535da483b6c54216f74f39edc18ce6a839898c0a/homeassistant/helpers/discovery.py#L163

Reply:
That's the setup of the platform entity integration, not the integration that implements the platform.

Reply:
Oh but that can't happen anymore since all the legacy platforms load in a task

Reply:
There are two places though that I'm not sure about

```
homeassistant/components/time_date/config_flow.py:from homeassistant.setup import async_prepare_setup_platform
homeassistant/components/time_date/config_flow.py:    platform = await async_prepare_setup_platform(hass, {}, SENSOR_DOMAIN, DOMAIN)
homeassistant/components/mailbox/__init__.py:from homeassistant.setup import async_prepare_setup_platform
homeassistant/components/mailbox/__init__.py:        platform = await async_prepare_setup_platform(hass, config, DOMAIN, p_type)
```

Reply:
I think mailbox can still deadlock
```
asterisk_cdr/mailbox.py		asterisk_mbox/mailbox.py	demo/mailbox.py
```

Reply:
https://github.com/home-assistant/core/blob/535da483b6c54216f74f39edc18ce6a839898c0a/homeassistant/components/mailbox/__init__.py#L120

Reply:
`time_date` isn't an issue

Reply:
https://github.com/home-assistant/core/pull/114774

Reply:
Yeah, I agree. Since we load the YAML platforms in tasks I can't see any deadlock anymore besides the (deprecated) mailbox integration.